app/gui/src/dashboard/providers/BackendProvider.tsx

```diff
@@ -110,15 +110,28 @@ export function useBackend(category: Category) {
   const remoteBackend = useRemoteBackend()
   const localBackend = useLocalBackend()
 
+  return pickBackend(category, remoteBackend, localBackend)
+}
+
+/**
+ * Pick the backend for the given category.
+ * @throws {Error} when a Local Backend is requested for a non-local project.
+ */
+function pickBackend(
+  category: Category,
+  remoteBackend: RemoteBackend,
+  localBackend: LocalBackend | null,
+) {
   if (isCloudCategory(category)) {
     return remoteBackend
-  } else {
-    invariant(
-      localBackend != null,
-      `This distribution of ${common.PRODUCT_NAME} does not support the Local Backend.`,
-    )
-    return localBackend
   }
+
+  invariant(
```

Comment:
also not changed in this PR but i do not think the user would appreciate a crash here

i would prefer logging in sentry but then resetting to a default cloud category (passed as an extra parameter)

Reply:
I assume that if we somehow get access to local backend when it's not available it's our fault and we should fail fast, I think